src/vehicle.cpp

```diff
@@ -3756,9 +3766,49 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
+        if(scoop_on){
+            operate_scoop();
+        }
+    }
+}
+void vehicle::operate_scoop()
+{
+    std::vector<int> scoops = all_parts_with_feature( "SCOOP" );
+    auto veh_points = get_points();
+    for( int scoop : scoops ) {
+        int chance_to_damage_item = parts[scoop].info().dmg_mod;
+        int max_pickup_size = parts[scoop].info().bonus;
+        const char *sound_msgs[] = {"Whirrrr", "Ker-chunk", "Swish", "Cugugugugug"};
+        sounds::sound( global_pos3() + parts[scoop].precalc[0], rng( 20, 35 ), sound_msgs[rng( 0, 3 )] );
+        for( const tripoint &position : veh_points ) {
+            g->m.mop_spills( position );
+            if( g->m.has_items( position ) ) {
+                item* that_item_there = NULL;
+                const map_stack q = g->m.i_at( position );
+                size_t itemdex = 0;
+                for( auto it : q ) {
+                    if( it.volume() < max_pickup_size ) {
+                        break;
+                    }
+                    itemdex++;
+                }
+                that_item_there = g->m.item_from( position, itemdex );
+                if( one_in( chance_to_damage_item ) &&
+                   that_item_there->damage < 4){//The scoop will not destroy the item, but it may damage it a bit.
+                        that_item_there->damage += 1;
+                        sounds::sound( position, 50, _("BEEEThump") );//The scoop gets a lot louder when breaking an item.
+                }
+                const int battery_deficit = discharge_battery( that_item_there->weight() * scoop_epower / rng( 8, 15 ) );
+                if( battery_deficit != 0
```

Comment:
Should probably be `== 0`, at the least description of the `discharge_battery` says so. Unless you've tested it and it works as expected at current settings.